skorch/callbacks/logging.py

```diff
@@ -728,3 +728,147 @@ def on_epoch_end(self, net, **kwargs):
+    >>> from skorch.callbacks.scoring import EpochScoring
+    >>> from skorch import NeuralNetClassifier
+    >>> from skorch.toy import make_classifier
+
+    >>> ex = Experiment()
+
+    >>> @ex.config
+    >>> def my_config():
+    ...     max_epochs = 20
+    ...     lr = 0.01
+
+    >>> X, y = make_classification()
+    >>> X, y = X.astype(np.float32), y.astype(np.int64)
+
+    >>> @ex.automain
+    >>> def main(_run, max_epochs, lr):
+    ...     # Take care to add additional scoring callbacks *before* the logger.
+    ...     net = NeuralNetClassifier(
+    ...         make_classifier(),
+    ...         max_epochs=max_epochs,
+    ...         lr=0.01,
+    ...         callbacks=[EpochScoring("f1"), SacredLogger(_run)]
+    ...     )
+    ...     # now fit your estimator to your data
+    ...     net.fit(X, y)
+
+    Then call this from the command line, e.g. like this:
+    ``python sacred-script.py with max_epochs=15``
+
+    You can also change other options on the command line and optionally
+    specify a backend.
+
+
+    Parameters
+    ----------
+    experiment : sacred.Experiment
+      Instantiated ``Experiment`` class.
+
+    log_on_batch_end : bool (default=False)
+      Whether to log loss and other metrics on batch level.
+
+    log_on_epoch_end : bool (default=True)
+      Whether to log loss and other metrics on epoch level.
+
+    batch_suffix : str (default=None)
+      A string that will be appended to all logged keys. By default (if set to
+      ``None``) "_batch" is used if batch and epoch logging are both enabled
+      and no suffix is used otherwise.
+
+    epoch_suffix : str (default=None)
+      A string that will be appended to all logged keys. By default (if set to
+      ``None``) "_epoch" is used if batch and epoch logging are both enabled
+      and no suffix is used otherwise.
+
+    keys_ignored : str or list of str (default=None)
+      Key or list of keys that should not be logged to Sacred. Note that in
+      addition to the keys provided by the user, keys such as those starting
+      with 'event_' or ending on '_best' are ignored by default.
+
+
+    .. _upstream: https://github.com/IDSIA/sacred#installing
+    """
+
+    def __init__(
+        self,
+        experiment,
+        log_on_batch_end=False,
+        log_on_epoch_end=True,
+        batch_suffix=None,
+        epoch_suffix=None,
+        keys_ignored=None,
+    ):
+        self.experiment = experiment
+        self.log_on_batch_end = log_on_batch_end
+        self.log_on_epoch_end = log_on_epoch_end
+        self.batch_suffix = batch_suffix
+        self.epoch_suffix = epoch_suffix
+        if self.batch_suffix is None:
+            self.batch_suffix = (
+                "_batch" if log_on_batch_end and log_on_epoch_end else ""
+            )
+        if self.epoch_suffix is None:
+            self.epoch_suffix = (
+                "_epoch" if log_on_batch_end and log_on_epoch_end else ""
+            )
```

Comment:
Could you please move this logic to `initialize`? Also, add an underscore to the attribute name: `self.batch_suffix_ = ...` (analog to `keys_ignored_`).

The reason is that all `__init__` parameters should be left untouched. That way, if a user calls `set_params(callbacks__sacred__batch_suffix="foo")`, this extra logic is honored, otherwise, it would be ignored. I know this is a total edge case, but we should be consistent with the general sklearn mechanics here.

Reply:
Of course, I didn't realize that the Loggers have to adhere to sklearn's Estimator API as well. Thanks for catching! Please take another look.

Reply:
Yes, we built it that way to allow to grid search even on callback parameters.